components/public-api-server/pkg/apiv1/workspace.go

```diff
@@ -403,6 +444,21 @@ func convertWorkspaceInstance(wsi *protocol.WorkspaceInstance, shareable bool) (
 		ports = append(ports, port)
 	}
 
+	// Calculate initial workspace folder location
+	var recentFolders []string
+	location := ""
+	if config != nil {
+		location = config.WorkspaceLocation
+		if location == "" {
+			location = config.CheckoutLocation
+		}
+	}
+	if location == "" && wsCtx != nil && wsCtx.Repository != nil {
+		location = wsCtx.Repository.Name
+
+	}
+	recentFolders = append(recentFolders, filepath.Join("/workspace", location))
```

Comment:
From what I understood reading the code, the server doesn't know about `/workspace` base folder, instead it's harcoded inside ws-manager, is it fine to harcode it here too?

Reply:
What's the ultimate source of truth for this information? If it's ws-manager, we should probably update ws-manager to report that directory also such that there's a single source of truth

Reply:
it's harcoded in many places, unsure which one is the source of truth, maybe workspace team can point it out cc @kylos101

Reply:
@jeanp413 I would consider the fact that this is hard coded to be non-blocking for this PR, as this is hard coded in webapp, workspace, and IDE components.

[webapp](https://github.com/gitpod-io/gitpod/blob/badbe674482c864b946eda4d672d9efd96fced27/components/server/src/workspace/config-provider.ts#L38)

[IDE](https://github.com/gitpod-io/gitpod/blob/badbe674482c864b946eda4d672d9efd96fced27/components/supervisor/pkg/terminal/service.go#L36)

[workspace](https://github.com/gitpod-io/gitpod/blob/badbe674482c864b946eda4d672d9efd96fced27/components/ws-manager-mk2/controllers/create.go#L39)